pkg/descheduler/node/node.go

```diff
@@ -96,32 +99,91 @@ func IsReady(node *v1.Node) bool {
 	return true
 }
 
-// PodFitsAnyOtherNode checks if the given pod fits any of the given nodes, besides the node
-// the pod is already running on. The node fit is based on multiple criteria, like, pod node selector
-// matching the node label (including affinity), the taints on the node, and the node being schedulable or not.
-func PodFitsAnyOtherNode(pod *v1.Pod, nodes []*v1.Node) bool {
+// NodeFit returns true if the provided pod can probably be scheduled onto the provided node.
+// This function is used when the NodeFit pod filtering feature of the Descheduler is enabled.
+// This function currently considers a subset of the Kubernetes Scheduler's predicates when
+// deciding if a pod would fit on a node, but more predicates may be added in the future.
+func NodeFit(nodeIndexer podutil.GetPodsAssignedToNodeFunc, pod *v1.Pod, node *v1.Node) []error {
+	// Check node selector and required affinity
+	var errors []error
+	ok, err := utils.PodMatchNodeSelector(pod, node)
+	if err != nil {
+		errors = append(errors, err)
+	} else if !ok {
+		errors = append(errors, fmt.Errorf("pod node selector does not match the node label"))
+	}
+	// Check taints (we only care about NoSchedule and NoExecute taints)
+	ok = utils.TolerationsTolerateTaintsWithFilter(pod.Spec.Tolerations, node.Spec.Taints, func(taint *v1.Taint) bool {
+		return taint.Effect == v1.TaintEffectNoSchedule || taint.Effect == v1.TaintEffectNoExecute
+	})
+	if !ok {
+		errors = append(errors, fmt.Errorf("pod does not tolerate taints on the node"))
+	}
+	// Check if the pod can fit on a node based off it's requests
+	ok, reqErrors := FitsRequest(nodeIndexer, pod, node)
```

Comment:
is `FitsRequest` called anywhere else? Wondering the reasoning behind making it its own exported function